### PR TITLE
Creatkeebs

### PR DESCRIPTION
**Pull request by Timliuzhaolu**
## Description

Add thera from creatkeebs.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/creatkeebs/thera/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#pragma once
```
Comment (drashna) [inline keyboards/creatkeebs/thera/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/creatkeebs/thera/config.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/creatkeebs/thera/config.h:2]:
```suggestion
```

File: keyboards/creatkeebs/thera/config.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
+
```
Comment (drashna) [inline keyboards/creatkeebs/thera/config.h:47]:
```suggestion
```

File: keyboards/creatkeebs/thera/config.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/creatkeebs/thera/config.h:50]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/creatkeebs/thera/info.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (drashna) [inline keyboards/creatkeebs/thera/info.json:1]: This is a via json, and as such does not belong in this repo.

File: keyboards/creatkeebs/thera/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,116 @@
+#include "kb.h"
```
Comment (drashna) [inline keyboards/creatkeebs/thera/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/creatkeebs/thera/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[7] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```
Comment (drashna) [inline keyboards/creatkeebs/thera/keymaps/default/keymap.c:67]:
```suggestion
```

File: keyboards/creatkeebs/thera/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+	}
+
+	if (usb_led & (1 << USB_LED_KANA)) {
+
+	} else {
+
+	}
+
+}
```
Comment (drashna) [inline keyboards/creatkeebs/thera/keymaps/default/keymap.c:116]:
```suggestion
```

File: keyboards/creatkeebs/thera/readme.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+* Ti badge
+* PVD weight (upgrade available)
+
+### Additional Information
+* Keyboard Maintainer: [Tim](https://github.com/Timliuzhaolu/qmk_firmware)
+* Hardware Supported: thera PCB
+* Hardware Availability: [ESCHIT.com](https://eschit.com/products/gb-thera75)
+
+### Building the Firmware
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/creatkeebs/thera/readme.md:18]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/creatkeebs/thera/rules.mk [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/creatkeebs/thera/rules.mk:57]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

Also, via should not be enabled at the keyboard level.

File: keyboards/creatkeebs/thera/thera.c [hunk 11]
```diff
@@ -0,0 +1 @@
+#include "thera.h"
```
Comment (drashna) [inline keyboards/creatkeebs/thera/thera.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/creatkeebs/thera/thera.h [hunk 12]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (drashna) [inline keyboards/creatkeebs/thera/thera.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/creatkeebs/thera/thera.h [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+#define KB_H
```
Comment (drashna) [inline keyboards/creatkeebs/thera/thera.h:3]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/creatkeebs/thera/info.json (hunk 5 above)
Comment (Timliuzhaolu) [reply keyboards/creatkeebs/thera/info.json:1]: Sorry about it, I have changed it now!

File: keyboards/creatkeebs/thera/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,14 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/creatkeebs/thera/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/creatkeebs/thera/keymaps/default/keymap.c (hunk 14 above)
Comment (Timliuzhaolu) [reply keyboards/creatkeebs/thera/keymaps/default/keymap.c:17]: Yeah, it has been added now. Thanks for pointing it out!

**Review by drashna — APPROVED** [review]

File: keyboards/creatkeebs/thera/readme.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
+# thera
+A 75% keyboard
+
+![thera](https://i.imgur.com/7w7ARcP.jpg)
```
Comment (fauxpark) [inline keyboards/creatkeebs/thera/readme.md:4]:
```suggestion
![thera](https://i.imgur.com/7w7ARcPh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Timliuzhaolu** [comment]
Hi both, is there anything else I have to do before merging? Please let me know. Thanks for all the helps and support! @drashna @fauxpark

**Review by fauxpark — APPROVED** [review]

**Comment by Timliuzhaolu** [comment]
Hi Ryan, will this be merged since I have got two approvals now? Thanks! @fauxpark

**Comment by drashna** [comment]
Thanks!

**Pull request merged**